src/System.Diagnostics.Process/src/System/Diagnostics/Process.cs

```diff
@@ -72,8 +72,10 @@ public partial class Process : IDisposable
         public event DataReceivedEventHandler ErrorDataReceived;
 
         // Abstract the stream details
-        internal AsyncStreamReader _output;
-        internal AsyncStreamReader _error;
+        private AsyncStreamReader _output;
+        private CancellationTokenSource _outputCts;
```

Comment:
As an alternative to having these two fields here, you could instead just add the _cts field to the AsyncStreamReader and construct the cts in the AsyncStreamReader's ctor, then expose a Cancel() method that just calls _cts.Cancel.  That saves the duplication here, and addresses my previous comment about passing around the CancellationToken.